modules/health_quest/spec/services/appointment_service_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe HealthQuest::AppointmentService do
+  subject { described_class.new(user) }
+
+  let(:user) { build(:user, :health_quest) }
+
+  before do
+    Flipper.enable('show_healthcare_experience_questionnaire')
+    allow_any_instance_of(HealthQuest::UserService).to receive(:session).and_return('stubbed_token')
+  end
+
+  describe '#get_appointment_by_id' do
+    let(:appointment) { subject.get_appointment_by_id(132) }
+
+    it 'is a Hash' do
+      expect(appointment).to be_a(Hash)
+    end
+
+    it 'has a data key of type OpenStruct' do
+      expect(appointment[:data]).to be_a(OpenStruct)
+    end
+
+    it 'data has relevant keys' do
+      expect(appointment[:data].respond_to?(:id)).to be(true)
```

Comment:
```suggestion
      expect(appointment[:data].key?(:id)).to be(true)
```

The test name suggests `appointment[:data]` is a hash rather than an object.

Would `.key?` be more appropriate here? Or should the test be renamed?

Reply:
Good catch. I think I will rename the test. `appointment[:data]` returns an OpenStruct object so sending it `respond_to?` seems a better fit.